controllers/cluster_controller.go

```diff
@@ -120,15 +121,33 @@ func (r *ClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr e
 		return ctrl.Result{}, err
 	}
 
-	if !cluster.ObjectMeta.DeletionTimestamp.IsZero() {
-		return r.reconcileDelete(ctx, cluster)
+	return ctrl.Result{}, nil
+}
+
+// reconcile handles cluster reconciliation.
+func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster) (err error) {
```

Comment:
Let's change the signature to return `(ctrl.Result, error)`

Reply:
And hopefully get rid of the `RequeueAfterError`

Reply:
We'd still need to handle `RequeueAfterError` in here though, given that sub-calls can return it

Reply:
... and eliminate it from sub-calls